src/devtools/client/inspector/components/App.tsx

```diff
@@ -1,254 +1,142 @@
-              onClick={this.toggle3PaneMode}
-              collapsePaneTitle={"Toggle off the 3-pane inspector"}
-              expandPaneTitle={"Toggle on the 3-pane inspector"}
-            /> */}
-            <ul className="tabs-menu" role="tablist">
-              {tabs}
-            </ul>
-          </nav>
-          <div className="panels">{activePanel}</div>
-        </div>
-      </div>
-    );
-  }
-
-  render() {
-    const { initializedPanels, is3PaneModeEnabled } = this.props;
-
-    const inspector = gToolbox.getPanel("inspector");
-    let markupView, rulesView;
-    if (inspector && initializedPanels.includes("inspector")) {
-      markupView = <MarkupApp inspector={inspector} />;
-      if (is3PaneModeEnabled) {
-        rulesView = <RulesApp {...inspector.rules.getRulesProps()} />;
+      case "computedview": {
+        return <ComputedApp />;
+      }
+      case "layoutview": {
+        const layoutProps = {
+          ...inspector.getCommonComponentProps(),
+          ...inspector.boxModel.getComponentProps(),
+          showBoxModelProperties: true,
+        };
+        return <LayoutApp {...layoutProps} />;
       }
+      default:
+        return null;
     }
-
-    return (
-      <div
-        className="inspector-responsive-container theme-body inspector"
-        data-localization-bundle="devtools/client/locales/inspector.properties"
-      >
-        <div id="inspector-splitter-box" ref={this.sidebarContainerRef}>
-          <SplitBox
-            ref={this.splitBoxRef}
-            className="inspector-sidebar-splitter"
-            initialSize={`${is3PaneModeEnabled ? prefs.sidebarSize : prefs.splitSidebarSize}px`}
-            minSize="20%"
-            maxSize="80%"
-            onMove={this.onSplitboxResize}
-            splitterSize={4}
-            endPanelControl={true}
-            startPanel={markupView}
-            endPanel={
-              <SplitBox
-                ref={this.sidebarSplitboxRef}
-                initialSize={`${is3PaneModeEnabled ? prefs.splitSidebarSize : 0}px`}
-                minSize={is3PaneModeEnabled ? "20%" : "0"}
-                maxSize="80%"
-                onMove={this.onSidebarSplitboxResize}
-                splitterSize={is3PaneModeEnabled ? 8 : 0}
-                endPanelControl={is3PaneModeEnabled}
-                startPanel={
-                  <div className="devtools-inspector-tab-panel">
-                    <div id="inspector-rules-container">
-                      <div id="inspector-rules-sidebar" hidden={false}>
-                        {rulesView}
-                      </div>
-                    </div>
-                  </div>
-                }
-                endPanel={
-                  <div className="devtools-inspector-tab-panel">
-                    <div id="inspector-sidebar-container">
-                      <div id="inspector-sidebar" hidden={false}>
-                        {this.renderSidebar()}
+  }, [inspector, inspectorInited, activeTab]);
```

Comment:
This array should include all the props that you pass to the components in the memoized function. Right now these props never change their value (I think), but there's no guarantee that will always be the case.

Reply:
Yeah, this is all, and better safe than sorry.